sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp

```diff
@@ -66,7 +70,40 @@ namespace Azure { namespace Core { namespace Http {
      * @brief Destructor.
      * @detail Cleans up CURL (invokes `curl_easy_cleanup()`).
      */
-    ~CurlConnection() { curl_easy_cleanup(this->m_handle); }
+    ~CurlConnection()
```

Comment:
I understand reproducing the underlying issue is difficult here. 

Is there any way to add simple unit tests that repro the issue to capture regressions?

Reply:
it should be a way, for sure.
It would be a long test, anyway.
We would need to open a connection and let it wait until server ends the connection so we can then validate there is no SIGPIPE signal upon cleaning the connection